examples/random.h

```diff
@@ -0,0 +1,31 @@
+#if defined(_WIN32)
+#include <bcrypt.h>
+#elif defined(__linux__)
+#include <fcntl.h>
+#include <unistd.h>
+#else
+#error "Couldn't identify the OS"
```

Comment:
Seems like this would happen on many common OSes (macOS, BSDs, ...). How about not compiling the examples by default?

Reply:
I skipped BSD because I didn't think anyone is really using it, but forgot about OSX.
I'll look into adding them both, see how compilcated it gets (I want to avoid something like https://github.com/bitcoin/bitcoin/blob/master/src/random.cpp and only support mostly recent OS versions)

Reply:
Okay, I hadn't read this conversation here before. Maybe it's really better not to compile the examples in our build system. If the concern is CI, we can compile them in the CI. If we want, we could add a super simple standalone Makefile that's entirely decoupled from our autoconf stuff.

Reply:
I'm trying to redo the travis and at least add MacOS to the CI
hmm stand alone make can be interesting, should that compile libsecp by it self? 
I can write something simple.

Otherwise we can still compile it in our CI, turn it off by default and bump our travis to bionic

Reply:
Added macOS and BSD, and will add macOS to the CI soon so we could actually test this.
rn they all use /dev/urandom, but if we bump to bionic I can replace with getrandom/getentropy

Reply:
I think the advantage of covering anything other than Linux for now is small compared to the maintenance overhead and bikeshedding. I'd suggest to do Linux only, use /dev/urandom, add comment to use `getrandom` if available, not compile by default, but add a travis test. Easy to revisit this later if there is demand for other systems but most important is the code, and not that it compiles on every user's system.

Reply:
Oh I didn't see your comment here. I suggested Linux/MacOS/Windows here:
https://github.com/bitcoin-core/secp256k1/issues/749#issuecomment-623964688 But I'd also be okay with just Linux. I think the main point is that we good documentation.

And I have a somewhat strong opinion on `/dev/urandom` vs `getrandom`. The issue with `/dev/urandom` is that it never blocks, i.e., it doesn't block at early boot when there has never been enough entropy. That's rare for Linux on the desktop but may be the case for a lightweight Linux on some device. So I'd really prefer `getrandom` and a comment to use `/dev/urandom` if `getrandom` is not available. If we don't compile the examples, then it does not matter if it's not available.

As I said, I'll try to find some refs. If you want, go ahead with this PR and I'll open a PR on top of it.

Reply:
Since we're updating travis to bionic in #750 (which I support), what you're suggesting sounds good to me (`getrandom` or bust) and test that in travis.

Reply:
Done :)

Reply:
Fixed, And added explanations about the different random sources for each platform